### DISCUSSION TITLE
Use HTTP connections WebSocket or others in App

### DISCUSSION THREAD

**Discussion by bsjaramillo** [vercel/community#426] [Show & Tell]
I have an application that connects to diferents websockets, but these use the HTTP, and i can't use HTTP over HTTPS so how i could do this ?

**Comment by Pieparker** [comment]
Sounds to me like a question for the web socket provider!

Does simply requesting the HTTP resource not work?

You may be able to [rewrite](https://vercel.com/docs/cli#project-configuration/rewrites) the web socket path (e.g. http://websocket-example.net/api/socket) to /api/socket on your domain). I've not tried this myself however it would be a worthwhile exercise.

**Reply by bsjaramillo** [reply]
Mmm yeah the socket provider works only under HTTP (and it doesn't belong me), and the hostings in vercel wokrs with HTTPS and it's not possible create a connection to the socket HTTP under HTTPS.

**Reply by Pieparker** [reply]
Have you checked whether a rewrite respects the ‘http’ protocol?
